std/assembly/typedarray.ts

```diff
@@ -68,6 +69,10 @@ export class Int8Array extends TypedArray<i8> {
   forEach(callbackfn: (value: i8, index: i32, self: Int8Array) => void): void {
     FOREACH<Int8Array, i8>(this, callbackfn);
   }
+
+  reverse(): Int8Array {
```

Comment:
`reverse(): this`?

Reply:
I have avoided using `this` because I did not understand what the `this` type was for in AssemblyScript. I seem to remember some kind of issue with using it before.

Reply:
It mean same as in TS. It's just alias for instance's type.

Reply:
![image](https://user-images.githubusercontent.com/3761339/54042716-cce67300-4198-11e9-98a5-2b542d591dc4.png)

Yeah. I would normally use the `this` type in TypeScript coding. For some reason, my tooling doesn't like this.

Reply:
Looks like it compiles though. I will start using this feature now.

Reply:
```ts
reverse(): this {
    return REVERSE<this, i32>(this);
  }
```

Reply:
Please see the latest commit. This should be resolved.

Reply:
ok, great!

Reply:
Some context on `this` type: It's actually the exact same type as the class it is used in and doesn't automatially cast up to subclasses. So, using it in let's say `TypedArray<T>` would not work as expected, because a function inside of a subclass like `Uint8Array` would return `TypedArray<T>` still.